python/ray/air/_internal/tensorflow_utils.py

```diff
@@ -51,6 +51,12 @@ def convert_pandas_to_tf_tensor(
             # don't cast any of the series and continue.
             pass
 
+        # if the columns are `ray.data.extensions.tensor_extension.TensorArray`,
+        # the dtype will be `object`. In this case, we need to set the dtype to
+        # none, and use the automatic type casting of `tf.convert_to_tensor`.
+        if isinstance(dtype, object):
+            dtype = None
```

Comment:
Could this be moved up to the above `except TypeError` block?

Generally, this seems good as a temporary workaround, although I think that the ideal behavior would be tweaking the `find_common_type()` utility such that it falls back to finding a common dtype among the underlying scalar type of the tensor extension type, e.g.:
```python
dtypes = [
    dtype if not isinstance(col, TensorArray) else col.numpy_dtype
    for dtype, (_, col) in zip(df.dtypes, df.items())
]
dtype: np.dtype = np.find_common_type(df.dtypes, [])
```